mobile/views/Search/index.js

```diff
@@ -0,0 +1,132 @@
+// @flow
+import * as React from 'react';
+import { withTheme } from 'styled-components';
+import { View, StyleSheet, Dimensions, TextInput } from 'react-native';
+import { TabViewAnimated, SceneMap, TabBar } from 'react-native-tab-view';
+import { TabLabel, SearchBar, SearchView } from './style';
+import { Constants } from 'expo';
+import ThreadsSearchView from './ThreadsSearchView';
+import CommunitiesSearchView from './CommunitiesSearchView';
+import PeopleSearchView from './PeopleSearchView';
+
+const initialLayout = {
+  height: 0,
+  width: Dimensions.get('window').width,
+};
+
+type TabView = {
+  key: string,
+  title: string,
+};
+
+type State = {
+  searchString: ?string,
+  index: number,
+  routes: Array<TabView>,
+};
+
+type Props = {
+  theme: Object,
+};
+
+class Search extends React.Component<Props, State> {
+  searchInput: React.ComponentType<TextInput>;
+
+  state = {
+    searchString: '',
+    index: 0,
+    routes: [
+      { key: 'threads', title: 'Threads' },
+      { key: 'communities', title: 'Communities' },
+      { key: 'people', title: 'People' },
+    ],
+  };
+
+  renderThreadsRoute = () => {
+    const { index, routes } = this.state;
+    if (routes[index].key !== 'threads') return <SearchView />;
+    if (!this.state.searchString) return <SearchView />;
+    return <ThreadsSearchView queryString={this.state.searchString} />;
+  };
+
+  renderCommunitiesRoute = () => {
+    const { index, routes } = this.state;
+    if (routes[index].key !== 'communities') return <SearchView />;
+    if (!this.state.searchString) return <SearchView />;
+    return <CommunitiesSearchView queryString={this.state.searchString} />;
+  };
+
+  renderPeopleRoute = () => {
+    const { index, routes } = this.state;
+    if (routes[index].key !== 'people') return <SearchView />;
+    if (!this.state.searchString) return <SearchView />;
+    return <PeopleSearchView queryString={this.state.searchString} />;
+  };
+
+  handleIndexChange = (index: number) => this.setState({ index });
+
+  renderTabs = (props: any) => {
+    const active = this.state.index === props.navigationState.index;
+    return (
+      <TabBar
+        indicatorStyle={{
+          backgroundColor: '#000',
+          height: StyleSheet.hairlineWidth,
+        }}
+        renderLabel={props => (
+          <TabLabel active={active}>{props.route.title}</TabLabel>
+        )}
+        style={{
+          backgroundColor: '#fff',
+          paddingTop: Constants.statusBarHeight,
+        }}
+        useNativeDriver
```

Comment:
I don't know if this does anything with Expo since we can't link any native code

Reply:
Not sure, we'll find out!